### ISSUE TITLE
Consistency with other Serilog organisation projects

### ISSUE THREAD

**Issue by nblumhardt** [serilog/serilog-sinks-async#10]
* [x] Create a maintainers team for the repo
 * [x] Set Jezz's permissions in the team (waiting on organisation m'ship)
 * [x] Repository name format, should be serilog-sinks-async
 * [x] Pare it back to the one `BlockingCollection` implementation
 * [x] Move to the standard Serilog build/test scripts and target .NET Core
 * [x] Create a `dev` branch and set it as the repository default branch
 * [x] Unlist 1.0.0 from NuGet - should get started with a prerelease version number, we can use `1.0.1-dev-nnnnn` to ensure upgrades go smoothly
 * [x] Turn on an AppVeyor build and get publishing happening
 * [x] Update README.MD with correct AppVeyor badges

**Comment by jezzsantos** [comment]
@nblumhardt  Thanks heaps for taking this on Nick.
Let me know if I can help. I'd be happy to contribute in any way that helps.

Setting up AppVeyor should be simple, since we already have an `appveyor.yml` file in the root. You just have to add the project to serilog account, and its done. Publishing needs configuring though.

**Comment by nblumhardt** [comment]
Done! Please excuse the one mighty commit - the build scripts, .NET Core support and requisite switch to Xunit, etc., was somewhat intertwined. The diff is a bit hard to follow but I got everything across (apart from a couple of duplicated tests).

Installed from NuGet and voila!

![image](https://cloud.githubusercontent.com/assets/342712/17456151/f298adb8-5c10-11e6-8dce-b8b7382faa91.png)

The default branch in the repository is now `dev`, so that new pull requests will default to targeting it. The build scripts will append a `-dev-nnnnn` suffix (like [this one](https://www.nuget.org/packages/Serilog.Sinks.Async/1.0.1-dev-00001)) except when on `master`, where no suffix is appended. This means that after releases (merges to master) the `project.json` in `dev` needs to be manually bumped to the next _major.minor_ version.
